templates/federated_reporting/transport.sh

```diff
@@ -21,28 +33,108 @@ if ! type "$CFE_FR_TRANSPORTER" >/dev/null; then
   exit 1
 fi
 
-dump_files="$(ls -1 "$CFE_FR_TRANSPORT_DIR/"*".sql.$CFE_FR_COMPRESSOR_EXT" 2>/dev/null)" ||
-  {
-    log "No files to transport."
-    exit 0
-  }
+function push() {
+  # Runs on the feeder hub, pushes dumps to the superhub.
+
+  dump_files="$(ls -1 "$CFE_FR_TRANSPORT_DIR/"*".sql.$CFE_FR_COMPRESSOR_EXT" 2>/dev/null)" ||
+    {
+      log "No files to transport."
+      exit 0
+    }
+
+  log "Transporting files: $dump_files"
+  some_failed=0
+  for dump_file in $dump_files; do
+    failed=0
+    mv "$dump_file" "$dump_file.transporting"
+    "$CFE_FR_TRANSPORTER" "$CFE_FR_TRANSPORTER_ARGS" "$dump_file.transporting" "$CFE_FR_SUPERHUB_LOGIN:$CFE_FR_SUPERHUB_DROP_DIR/$(basename "$dump_file")" &&
+      rm -f "$dump_file.transporting" || failed=1
+    if [ "$failed" != 0 ]; then
+      log "Transporting file $dump_file failed!"
+      some_failed=1
+    fi
+  done
+
+  if [ "$some_failed" != "0" ]; then
+    log "Transporting files: FAILED"
+    return 1
+  else
+    log "Transporting files: DONE"
+    return 0
+  fi
+}
+
+function pull() {
+  # $@ -- feeder hubs to pull the dumps from
+  feeder_lines="$(printf "%s\n" "$@")"
+  log "Pulling dumps from: $feeder_lines"
+
+  chmod u+x "$(dirname "$0")/pull_dumps_from.sh"
+
```

Comment:
shouldn't perms like this be managed by policy? seems like this is a good safety measure but I wonder if it is noise.

Reply:
I was wondering about the same. But then I thought this won't hurt. And it happened to me while testing that the file was not executable (copied over by `cf-remote`).